src/pages/maintenance/Maintenance.js

```diff
@@ -1,180 +1,126 @@
+        Object.entries(checkboxes).forEach(([key, checked]) => {
+            formData.append(key, checked)
         })
+        return formData
     }
 
-    selectedCheckboxesCount() {
-        let selectedCheckboxes = 0
-        const checkboxKeys = Object.keys(this.state.checkboxes)
-        for (let i = 0; i < checkboxKeys.length; i++) {
-            const key = checkboxKeys[i]
-            const checked = this.state.checkboxes[key].checked
-            if (checked) {
-                selectedCheckboxes += 1
-            }
-        }
-        return selectedCheckboxes
-    }
-
-    areActionsDisabled() {
-        return this.props.loading
+    return {
+        checkboxes,
+        toggleCheckbox,
+        toFormData,
     }
+}
 
-    buildFormData() {
-        let formData = null
-        const checkboxKeys = Object.keys(this.state.checkboxes)
-        for (let i = 0; i < checkboxKeys.length; i++) {
-            const key = checkboxKeys[i]
-            const checked = this.state.checkboxes[key].checked
-            formData = formData || new FormData()
-            formData.append(key, checked)
-        }
-
-        return formData
-    }
+const Checkboxes = ({ checkboxes, toggleCheckbox, disabled }) => (
+    <div className={styles.checkboxes}>
+        {Object.entries(checkboxes).map(([key, checked]) => (
+            <Checkbox
+                key={key}
+                className={styles.checkbox}
+                label={i18nKeys.maintenance.checkboxes[key]}
+                checked={checked}
+                onChange={() => toggleCheckbox(key)}
+                disabled={disabled}
+            />
+        ))}
+    </div>
+)
 
-    performMaintenance = () => {
-        const api = this.context.d2.Api.getApi()
-        const formData = this.buildFormData()
+Checkboxes.propTypes = {
+    checkboxes: PropTypes.object.isRequired,
+    disabled: PropTypes.bool.isRequired,
+    toggleCheckbox: PropTypes.func.isRequired,
+}
 
-        if (formData) {
-            this.setLoadingPageState()
-            api.post(MAINTENANCE_ENDPOINT, formData)
-                .then(() => {
-                    if (this.isPageMounted()) {
-                        this.context.updateAppState({
-                            showSnackbar: true,
-                            snackbarConf: {
-                                type: SUCCESS,
-                                message: i18n.t(
-                                    i18nKeys.maintenance.actionPerformed
-                                ),
-                            },
-                            pageState: {
-                                loading: false,
-                            },
-                        })
-                    }
-                })
-                .catch(error => {
-                    if (this.isPageMounted()) {
-                        this.setLoadedPageWithErrorState(error)
-                    }
-                })
+const Maintenance = ({ sectionKey }) => {
```

Comment:
This file is also getting fairly large. I think splitting it up would help with the readability.